sdk/ai/azure-ai-client/samples/endpoints/async_samples/sample_endpoints_async.py

```diff
@@ -0,0 +1,141 @@
+# ------------------------------------
+
+"""
+FILE: sample_endpoints_async
+
+DESCRIPTION:
+    This sample demonstrates how to enumerate and get endpoints from an AzureAIClient.
+
+USAGE:
+    python sample_endpoints_async
+
+    Before running the sample:
+
+    pip install azure.ai.client aiohttp azure-identity
+
+    Set the environment variables with your own values:
+    1) AI_CLIENT_CONNECTION_STRING - the Azure AI Project connection string, as found in your AI Studio Project.
+"""
+
+import asyncio
+import os
+from azure.ai.client.aio import AzureAIClient
+from azure.ai.client.models import EndpointType, AuthenticationType
+from openai import AzureOpenAI
+from azure.ai.inference.aio import ChatCompletionsClient
+from azure.ai.inference.models import UserMessage
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+from azure.core.credentials import AzureKeyCredential
+
+async def sample_endpoints_async():
+
+    # Create an Azure AI Client from a connection string, copied from your AI Studio project.
+    # It should be in the format "<Endpoint>;<AzureSubscriptionId>;<ResourceGroup>;<WorkspaceName>"
+    async with AzureAIClient.from_connection_string(
+        credential=DefaultAzureCredential(),
+        connection=os.environ["AI_CLIENT_CONNECTION_STRING"],
+    ) as ai_client:
+
+    # Or, you can create the Azure AI Client by giving all required parameters directly
+    # async with AzureAIClient(
+    #     credential=DefaultAzureCredential(),
+    #     endpoint=os.environ["AI_CLIENT_ENDPOINT"],
+    #     subscription_id=os.environ["AI_CLIENT_SUBSCRIPTION_ID"],
+    #     resource_group_name=os.environ["AI_CLIENT_RESOURCE_GROUP_NAME"],
+    #     workspace_name=os.environ["AI_CLIENT_WORKSPACE_NAME"],
+    # ) as ai_client:
+
+        # List all endpoints of a particular "type", with or without their credentials:
+        print("====> Listing of all Azure Open AI endpoints:")
+        async for endpoint in ai_client.endpoints.list(
+            endpoint_type=EndpointType.AZURE_OPEN_AI,  # Optional. Defaults to all types.
+            populate_secrets=True,  # Optional. Defaults to "False"
+        ):
+            print(endpoint)
+
+        # Get the default endpoint of a particular "type" (note that since at the moment the service
+        # does not have a notion of a default endpoint, this will return the first endpoint of that type):
+        endpoint = await ai_client.endpoints.get_default(
+            endpoint_type=EndpointType.AZURE_OPEN_AI, populate_secrets=True  # Required.  # Optional. Defaults to "False"
+        )
+        print("====> Get default Azure Open AI endpoint:")
+        print(endpoint)
+
+        # Get an endpoint by its name:
+        endpoint = await ai_client.endpoints.get(
+            endpoint_name=os.environ["AI_CLIENT_CONNECTION_NAME"], populate_secrets=True  # Required.
+        )
+        print("====> Get endpoint by name:")
+        print(endpoint)
+
+        exit()
```

Comment:
Why there is exit() here? The code below does not ever run if I am looking correctly?

Reply:
Still need to polish the bottom half of this simple, it's work in progress (but should be functional). I'll take care of that in another PR.